docs/csharp/csharp-6.md

```diff
@@ -1,17 +1,570 @@
+interface:
+
+[!code-csharp[nameofNotify](../../samples/snippets/csharp/new-in-6/viewmodel.cs#nameofNotify)]
+
+The advantage of using the `nameof` operator over a constant string is
+that tools can understand the symbol. If you use refactoring tools to
+rename the symbol, it will rename it in the `nameof` expression. Constant
+strings don't have that advantage. Try it yourself in your favorite editor:
+Rename a variable, and any nameof expressions will update as well.
+
+The `nameof` expression produces the unqualified name of its argument
+(`LastName` in the examples above) even if you use the fully qualified
+name for the argument:
+
+[!code-csharp[QualifiedNameofNotify](../../samples/snippets/csharp/new-in-6/viewmodel.cs#QualifiedNameofNotify)]
+
+The above `nameof` expression produces `FirstName`, not `UXComponents.ViewModel.FirstName`.
+
+## Await in Catch and Finally blocks
+
+C# 5 had several limitations around where you could place `await` expressions.
+One of those has been removed in C# 6. You can now use `await` in `catch`
+or `finally` expressions. 
+
+The addition of await expressions in catch and finally blocks may appear
+to complicate how those are processed. Let's add an example to discuss
+how this appears. In any async method, you can use an await expression
+in a finally clause.
+
+With C# 6, you can also await in catch expressions. This is most often
+used with logging scenarios:
+
+[!code-csharp[AwaitFinally](../../samples/snippets/csharp/new-in-6/NetworkClient.cs#AwaitFinally)]
+
+The implementation details for adding `await` support inside `catch`
+and `finally` clauses ensures that the behavior is consistent with the
+behavior for synchronous code. When code executed in a `catch` or `finally`
+clause throws, execution looks for a suitable `catch` clause in the next
+surrounding block. If there was a current exception, that exception is
+lost. The same happens with `await`ed expressions in `catch` and `finally`
+clauses: a suitable `catch` is searched for, and the current exception,
+if any, is lost.  
+
+> [!NOTE]
+> This behavior is the reason it's recommended to write `catch` and `finally`
+> clauses carefully, to avoid introducing new exceptions.
+
+## Index Initializers
+
+***Index Initializers*** is one of two features that make collection
+initializers more consistent. In earlier releases of C#, you could use
+***collection initializers*** only with sequence style collections:
+
+[!code-csharp[ListInitializer](../../samples/snippets/csharp/new-in-6/initializers.cs#ListInitializer)]
+
+Now, you can also use them with `Dictionary` collections and similar types:
+
+[!code-csharp[DictionaryInitializer](../../samples/snippets/csharp/new-in-6/initializers.cs#DictionaryInitializer)]
+
+This features means that associative containers can be initialized using
+syntax similar to what's been in place for sequence containers for several
+versions.
+
+### Extension `Add` methods in collection initializers
```

Comment:
wondering if this scenario would benefit from having the full sample for people to see?